jdbc/src/ragtime/jdbc.clj

```diff
@@ -16,20 +16,29 @@
                         [[:id "varchar(255)"]
                          [:created_at "varchar(32)"]]))
 
-(defn- get-table-names* [conn]
+(defn- get-table-metadata* [conn]
   (-> conn
       (.getMetaData)
       (.getTables (.getCatalog conn) nil "%" nil)
-      (sql/metadata-result {:row-fn :table_name})))
+      (sql/metadata-result)))
 
-(defn- get-table-names [db-spec]
+(defn- get-table-metadata [db-spec]
   (if-let [conn (sql/db-find-connection db-spec)]
-    (get-table-names* conn)
+    (get-table-metadata* conn)
     (with-open [conn (sql/get-connection db-spec)]
-      (get-table-names* conn))))
+      (get-table-metadata* conn))))
+
+(defn- has-schema?
+  [table-name]
+  (re-find #"\." table-name))
 
 (defn- table-exists? [db-spec ^String table-name]
-  (some #(.equalsIgnoreCase table-name %) (get-table-names db-spec)))
+  (some (fn [row]
```

Comment:
Let's change this part to:

```clojure
(defn- table-exists? [db-spec table-name]
  (some (partial metadata-matches-table? table-name)
        (get-table-metadata db-spec))
```

And then write:

```clojure
(defn metadata-matches-table? [^String table-name metadata]
   (.equalsIgnoreCase table-name
                      (if (.contains table-name ".")
                        (str (:table_schem metadata) "." (:table_name metadata))
                        (:table_name metadata)))
```

Which means we can get rid of `has-schema?`